### PR TITLE
Shared Task pages

### LINKED ISSUE #612: Text2text task link redirects to text2text
I don't know if it should be hub that I'm opening the issue here, but basically we have only one task for both text generation and text2text generation (because they're pretty much the same) but the widget link redirects to non-existent text2text  task page. (You can see below) (The task name is "Text Generation" and it covers text2text as well)
@osanseviero @beurkinger 

Ekran Resmi 2022-01-20 18 41 28
a

### PR DESCRIPTION
**Pull request by beurkinger**
fix #612

- make text-generation ans text2text-generation share the same data
- change TaskData typing so we have a better view of which tasks have task pages, which ones don't, and which ones share the same task page
- remove link from ModelPipelineTag (as we don't necessarily want it everywhere) 
- add link to task page in WidgetWrapper
- don't display link if no task page exists

### CODE REVIEW

File: tasks/src/tasksData.ts [hunk 1]
```diff
@@ -18,20 +17,39 @@ import translation from "./translation/data";
...
-	"question-answering":           questionAnswering,
-	"sentence-similarity":          sentenceSimilarity,
-	"summarization":                summarization,
-	"text-classification":          textClassification,
-	"text-generation":              textGeneration,
-	"text-to-speech":               textToSpeech,
-	"token-classification":         tokenClassification,
-	"translation":                  translation,
+
+// To make comparisons easier, task order is the same as in /lib/interfaces/Types.ts
```
Comment (julien-c) [inline tasks/src/tasksData.ts:21]: 👍

File: js/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte [hunk 2]
```diff
@@ -22,7 +24,16 @@
 	class="flex items-center justify-between flex-wrap w-full max-w-full text-sm text-gray-500 mb-0.5"
 >
 	{#if pipeline}
-		<ModelPipelineTag classNames="mr-2 mb-1.5" {pipeline} />
+		<a
+			class={TASKS_DATA[pipeline] ? "hover:underline" : undefined}
+			href={TASKS_DATA[pipeline] ? `/tasks/${pipeline}` : undefined}
+			target="_blank"
+			title={TASKS_DATA[pipeline]
+				? "Learn more about {PipelineType[pipeline]}"
```
Comment (Pierrci) [inline js/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:32]:
```suggestion
				? `Learn more about ${PipelineType[pipeline]}`
```
(and you need to import `PipelineType` of course)

Comment (beurkinger) [reply js/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:32]: 🙈

Comment (beurkinger) [reply js/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte]: Done !

**Review by Pierrci — APPROVED** [review]

**Review by osanseviero — APPROVED** [review]
Thanks for this PR!

File: js/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte [hunk 3]
```diff
@@ -22,7 +28,14 @@
 	class="flex items-center justify-between flex-wrap w-full max-w-full text-sm text-gray-500 mb-0.5"
 >
 	{#if pipeline}
-		<ModelPipelineTag classNames="mr-2 mb-1.5" {pipeline} />
+		<a
+			class={TASKS_DATA[task] ? "hover:underline" : undefined}
```
Comment (Pierrci) [inline js/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:32]: the `hover:underline` doesn't seem to have any effect on firefox, but it's working on chrome, strange (no need to spend too much time on this though)

Comment (beurkinger) [reply js/src/lib/components/InferenceWidget/shared/WidgetHeader/WidgetHeader.svelte:32]: @Pierrci  I'll merge first (as some people are eager to get this PR out) then investigate :)

**Pull request merged**